src/plenigo/internal/services/Service.php

```diff
@@ -74,6 +76,8 @@ protected static function getRequest($endPoint, $oauth = false, array $params =
     protected static function postRequest($endPoint, $oauth = false, array $params = array())
     {
         if($oauth){
```

Comment:
This is the same code like above. I think you could externalize that to another method that creates the url.